more_itertools/more.py

```diff
@@ -949,7 +1105,35 @@ def walk(node, level):
     yield from walk(iterable, 0)
 
 
-def side_effect(func, iterable, chunk_size=None, before=None, after=None):
+@overload
+def side_effect(
```

Comment:
@ilai-deutel : I was taking a quick look at `side_effect` since @bbayles mentioned it was "un-aesthetic".  At a glance it is not obvious to me why the `@overloaded` versions are necessary. Perhaps you could add comment about it for the benefit of future readers of the code?

Reply:
Aha, I see - it is because you want to enforce that `chunk_size` is specified if and only if `func`  takes a `List` as its input.